### PR TITLE
Allow `.` in param/result names via subscript.

### PR DESCRIPTION
**Pull request by mattmoor**
This change allow folks to use `.` in parameter names (e.g. `dev.mattmoor.my-param`), and reference them via the subscript operator (e.g. `params["dev.mattmoor.my-param"]`) to avoid ambiguity introduced by the mixing of `.`s.

TEP: https://github.com/tektoncd/community/pull/503

/kind enhancement
/hold

Holding until the TEP is merged.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
Parameter and result names may now contain `.` and be referenced via the subscript operator (e.g. `$(params["foo.bar"])`
```

### CODE REVIEW

**Comment by mattmoor** [comment]
I'm sure there are some docs changes needed for this, although I'm not sure where those docs are, so if folks could provide pointers to any docs they think should be updated to reflect this change I'd appreciate it 🙏 

/kind feature

**Comment by mattmoor** [comment]
I'm going to try and expand this with e2e test coverage as well, so not checking the test box yet.

**Comment by mattmoor** [comment]
Only failure seems to be outside of what I've modified, so let's see if it repro's.

/retest

**Comment by mattmoor** [comment]
I think `docs/tasks.md` may be the place to update for the docs ✅ , looking at that now.

**Comment by mattmoor** [comment]
Ok, I updated `docs/tasks.md` and also found `docs/variables.md` and added a couple rows to the pertinent tables with the new syntax.  The prior commit also tweaked a few of the e2e tests to exercise the subscript syntax, so I've checked off "docs" and "e2e".  I'm ofc happy to expand both of these if folks have specific pointers to things I missed.

For the commit message, I've left this unchecked as I am somewhat reluctant to fork the TEP PR as the source of truth (esp. before it is merged), but if folks have a good example, I'd be happy to circle back and update things to follow a strong precedent.

**Comment by mattmoor** [comment]
/unhold

The TEP has merged

**Comment by vdemeester** [comment]
/hold cancel
Maybe ? 😛

**Comment by mattmoor** [comment]
`/unhold` works for Knative prow, so either we have some snowflake or y'all are on a really old version 😅 

Either way, thanks!

**Comment by vdemeester** [comment]
> `/unhold` works for Knative prow, so either we have some snowflake or y'all are on a really old version 
> 
> Either way, thanks!

Really old version it is 😉 😛 😅

**Comment by dlorenc** [comment]
/lgtm

nice!

File: pkg/reconciler/taskrun/resources/apply.go
Comment (vdemeester) [inline pkg/reconciler/taskrun/resources/apply.go:48]: Did I wrote that FIXME ? I think I meant "with deprecating v1alpha1", maybe ? 🤔 👴🏼

**Review by vdemeester — APPROVED** [review]
/hold
/cc @afrittoli @bobcatfish

**Comment by vdemeester** [comment]
(holding just a day or two or less to allow core team to review 👼🏼 )

File: pkg/reconciler/taskrun/resources/apply.go
Comment (mattmoor) [reply pkg/reconciler/taskrun/resources/apply.go:48]: Yeah, I just moved it 😁

Comment (vdemeester) [reply pkg/reconciler/taskrun/resources/apply.go:48]: yeah.. I saw that 😝 I am confused by my own comment 😹 damn…

**Comment by dlorenc** [comment]
Are we still waiting on anything here?

**Comment by vdemeester** [comment]
@dlorenc let's validate this in the upcoming API WG (today) 👼🏼

**Comment by mattmoor** [comment]
@vdemeester Just seeing this, but don't think it was discussed... 😅

**Comment by vdemeester** [comment]
> @vdemeester Just seeing this, but don't think it was discussed...

That's a good point 😅 
@tektoncd/core-maintainers last call for review, I'll check tomorrow morning before the release 😝

**Comment by mattmoor** [comment]
Just popped open my pending PR list and saw this (I'd forgotten about it 🤣 ), should we `/hold cancel` it?

**Comment by vdemeester** [comment]
/hold cancel
Indeed 😛

**Comment by bobcatfish** [comment]
Late to the party but I'm excited to see this going in! Next release is gonna be a good one :D

**Comment by bobcatfish** [comment]
That being said, one more process nit: technically the TEP should be "implementable" before the implementation gets merged (another opportunity for some automation to enforce?) - I added a comment to the TEP pr also https://github.com/tektoncd/community/pull/503/files#r709442151

**Comment by mattmoor** [comment]
I'm curious if there are examples of TEPs that have been merged as "proposed" and never been made "implementable"?  I'm sort of curious why this extra hop is in there, and whether it's been useful historically?

Happy to discuss elsewhere as this is sort of an odd place to have that discussion 🤣

**Comment by bobcatfish** [comment]
@mattmoor wouldn't be surprised - the process is documented at https://github.com/tektoncd/community/blob/main/teps/0001-tekton-enhancement-proposal-process.md#tep-workflow - looking at some of the history there might help; a great place to discuss changes would be the API working group and/or an issue/pr over there

**Comment by skaegi** [comment]
I know this PR is merged already, but I think it would make a lot of sense to support single quotes in addition to double quotes. This both matches how Kubernetes [references info with the downwards api](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api) and also how [bracket notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors#bracket_notation) is typically used.

**Pull request merged**
